src/Core/Grid/Definition/Factory/OrderReturnProductsGridDefinitionFactory.php

```diff
@@ -0,0 +1,255 @@
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Exception\OrderReturnProductException;
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\BulkActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Bulk\Type\SubmitBulkAction;
+use PrestaShop\PrestaShop\Core\Grid\Action\ModalOptions;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\RowActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Action\Row\Type\SubmitRowAction;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\ActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\BulkActionColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\DataColumn;
+use PrestaShop\PrestaShop\Core\Grid\Column\Type\Common\TemplateColumn;
+use PrestaShop\PrestaShop\Core\Grid\Filter\Filter;
+use PrestaShop\PrestaShop\Core\Grid\Filter\FilterCollection;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Type\SearchAndResetType;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Class OrderReturnGridDefinitionFactory builds grid definition for order returns grid.
+ */
+final class OrderReturnProductsGridDefinitionFactory extends AbstractFilterableGridDefinitionFactory
+{
+    use BulkDeleteActionTrait;
+    use DeleteActionTrait;
+
+    public const GRID_ID = 'order_return_products';
+
+    /**
+     * @var int
+     */
+    private $orderReturnId;
+
+    /**
+     * OrderReturnProductsGridDefinitionFactory constructor.
+     *
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param RequestStack $requestStack
+     *
+     * @throws OrderReturnProductException
+     */
+    public function __construct(
+        HookDispatcherInterface $hookDispatcher,
+        RequestStack $requestStack
+    ) {
+        parent::__construct($hookDispatcher);
+        $this->setOrderReturnId($requestStack);
+    }
+
+    /**
+     * Sets order return id directly from request attribute. On not found case throws exception.
+     *
+     * @param RequestStack $requestStack
+     *
+     * @throws OrderReturnProductException
+     */
+    private function setOrderReturnId(RequestStack $requestStack): void
```

Comment:
private method should go down bellow protected ones

Reply:
Btw is real bad practice to depend on request here. I know there is currently no other option, but I think we should alter the design of grid definition factory to allow adding some runtime parameters for such cases.
E.g. introduce optional $options param into GridFactory->getGrid($filters, array $options). This would also require same for GridDefinitionFactory->getDefinition($options = []). But I think its doable without BC breaks this way.
In the end in controller then you could call it like: 
```
        $gridFactory->getGrid(
            $filters,
            ['orderReturnId' => $request->request->get('orderReturnId')]
        );
```

Anyway I think this deserves a dedicated PR.
Wdyt @JevgenijVisockij, @jolelievre?

Reply:
we've with @jolelievre figured a way to integrate some options (or rather filters) during runtime in a much better way.
You should create dedicated GridFactory and a dedicated filters builder, to pass orderReturnId into $filters.
See example in this PR with feature value https://github.com/PrestaShop/PrestaShop/pull/32334/files#diff-45667b705fa4b87859bc55a90391bec7aa547b2e72dba20d589b0d9dc6d68047R112